core/src/main/scala/com.snowplowanalytics.stream/loader/StreamLoaderApp.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2014-2017 Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache
+ * License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.
+ *
+ * See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.stream.loader
+
+// Java
+import java.io.File
+
+// Config
+import com.typesafe.config.ConfigFactory
+
+// Scala
+import scopt.OptionParser
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Pureconfig
+import pureconfig._
+
+// This project
+import sinks._
+import model._
+
+/**
+ * Main entry point for the Elasticsearch sink
+ */
+trait StreamLoaderApp extends App {
+
+  val executor: Validation[String, Runnable]
+  lazy val arguments = args
+
+  private def parseConfig(): Option[StreamLoaderConfig] = {
+    val projectName = "snowplow-stream-loader"
+    case class FileConfig(config: File = new File("."))
+    val parser:OptionParser[FileConfig] = new scopt.OptionParser[FileConfig](projectName) {
+      head(projectName, com.snowplowanalytics.stream.loader.generated.Settings.version)
+      help("help")
+      version("version")
+      opt[File]("config").required().valueName("<filename>")
+        .action((f: File, c: FileConfig) => c.copy(config = f))
+        .validate(f =>
+          if (f.exists) success
+          else failure(s"Configurationfile $f does not exist")
+        )
+    }
+
+    val config = parser.parse(arguments, FileConfig()) match {
+      case Some(c) => ConfigFactory.parseFile(c.config).resolve()
+      case None    => ConfigFactory.empty()
+    }
+
+    if (config.isEmpty()) {
+      System.err.println("Empty configuration file")
+      System.exit(1)
+    }
+
+    implicit def hint[T] = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
+    implicit val queueHint = new FieldCoproductHint[Queue]("enabled")
+
+    val streamLoaderConf = loadConfig[StreamLoaderConfig](config) match {
+      case Left(e) =>
+        System.err.println(s"configuration error: $e")
+        System.exit(1)
+        None
+       case Right(c) => Some(c)
+    }
+
+    streamLoaderConf
+  }
+
+  val config:StreamLoaderConfig = parseConfig().get
+
+  val tracker = config.monitoring.map(e => SnowplowTracking.initializeTracker(e.snowplow))
+
+  val goodSink = config.sink.good match {
+    case "stdout" => Some(new StdouterrSink)
+    case "elasticsearch" => None
```

Comment:
This can't be part of the core since it's what you're trying to abstract over.

You can look at [stream-enrich](https://github.com/snowplow/snowplow/tree/master/3-enrich/stream-enrich) to know how we've abstracted over the streaming platform.

Reply:
I will move it `es` module since core doesn't need for any reason yet. Having it like `stream-enrich` is a long shot for this ticket.